packages/bun/src/integrations/bunserver.ts

```diff
@@ -0,0 +1,143 @@
+import { captureException, getCurrentHub, runWithAsyncContext, startSpan, Transaction } from '@sentry/core';
+import type { Integration } from '@sentry/types';
+import {
+  addExceptionMechanism,
+  getSanitizedUrlString,
+  objectify,
+  parseUrl,
+  tracingContextFromHeaders,
+} from '@sentry/utils';
+
+function sendErrorToSentry(e: unknown): unknown {
+  // In case we have a primitive, wrap it in the equivalent wrapper class (string -> String, etc.) so that we can
+  // store a seen flag on it.
+  const objectifiedErr = objectify(e);
+
+  captureException(objectifiedErr, scope => {
+    scope.addEventProcessor(event => {
+      addExceptionMechanism(event, {
+        type: 'bun',
+        handled: false,
+        data: {
+          function: 'serve',
+        },
+      });
+      return event;
+    });
+
+    return scope;
+  });
+
+  return objectifiedErr;
+}
+
+/**
+ * Instruments `Bun.serve` to automatically create transactions and capture errors.
+ */
+export class BunServer implements Integration {
+  /**
+   * @inheritDoc
+   */
+  public static id: string = 'BunServer';
+
+  /**
+   * @inheritDoc
+   */
+  public name: string = BunServer.id;
+
+  /**
+   * @inheritDoc
+   */
+  public setupOnce(): void {
+    instrumentBunServe();
+  }
+}
+
+/**
+ * Instruments Bun.serve by patching it's options.
+ */
+export function instrumentBunServe(): void {
+  Bun.serve = new Proxy(Bun.serve, {
```

Comment:
dayum we're lucky this is just a field on a global - let's pray this never changes 🙏

Reply:
I don't think they'll break this for a while, but let's see.